src/compiler/checker.ts

```diff
@@ -3690,6 +3697,48 @@ module ts {
             }
             return links.resolvedType;
         }
+        
+        function getTypeGuardType(node: TypeGuardTypeNode): Type {
+            let type = <TypeGuardType>createType(TypeFlags.TypeGuard);
+            let hasParameters = false;
+            type.parameterIndex = -1;
+            if (node.target) {
+                let functionDeclaration = <SignatureDeclaration>node.target.parent.parent;

```

Comment:
I'm not sure I understand here either? This block of code is for checking that the declared TypeGuardType has a target that corresponds to a declared parameter in a signature.

I could change the variable name to just `signature`.